### PR TITLE
Remove R script/notebook package installs from remaining modules (Part 4 of 4) 

### LINKED ISSUE #690: Remove any R package installation from notebooks or scripts, especially with BiocManager
Context: #689 - which is adding a script to specifically install Bioconductor packages on the project Docker container.

Having logic that checks whether or not a package is installed and if not installs it is reasonable if someone were to run an analysis outside the Docker container. But if we were to miss something in the Docker image and then run an analysis it could end up unpinning versions (as discovered on the journey of #684). So, we should remove any package installation from notebooks or scripts, especially any installation via `BiocManager`.

Here's a list from @sjspielman:

```
focal-cn-file-preparation/04-prepare-cn-file.R:  install.packages("BiocManager")
focal-cn-file-preparation/04-prepare-cn-file.R:  BiocManager::install("GenomicRanges", update = FALSE)
focal-cn-file-preparation/04-prepare-cn-file.R:  BiocManager::install("IRanges", update = FALSE)
focal-cn-file-preparation/04-prepare-cn-file.R:  BiocManager::install("annotatr", update = FALSE)
focal-cn-file-preparation/04-prepare-cn-file.R:  BiocManager::install("TxDb.Hsapiens.UCSC.hg38.knownGene", update = FALSE)
focal-cn-file-preparation/04-prepare-cn-file.R:  BiocManager::install("org.Hs.eg.db", update = FALSE)
focal-cn-file-preparation/04-prepare-cn-file.R:  BiocManager::install("AnnotationDbi", update = FALSE)
gene-set-enrichment-analysis/01-conduct-gsea-analysis.R:## Load and/or install libraries ##
gene-set-enrichment-analysis/01-conduct-gsea-analysis.R:  install.packages("msigdbr")
gene-set-enrichment-analysis/01-conduct-gsea-analysis.R:  install.packages("BiocManager")
gene-set-enrichment-analysis/01-conduct-gsea-analysis.R:  BiocManager::install("GSVA")
molecular-subtyping-ATRT/02-ATRT-molecular-subtyping-plotting.R:  install.packages("ComplexHeatmap")
molecular-subtyping-ATRT/02-ATRT-molecular-subtyping-plotting.R:  install.packages("matrixStats")
molecular-subtyping-ATRT/02-ATRT-molecular-subtyping-plotting.R:  install.packages("ggfortify")
oncoprint-landscape/01-plot-oncoprint.R:  install.packages("BiocManager")
oncoprint-landscape/01-plot-oncoprint.R:  BiocManager::install("maftools")
cnv-chrom-plot/gistic_plot.Rmd:  install.packages("BiocManager")
cnv-chrom-plot/gistic_plot.Rmd:  BiocManager::install("ggbio")
cnv-comparison/01-cnv-comparison-plotting.Rmd:  install.packages("BiocManager")
cnv-comparison/01-cnv-comparison-plotting.Rmd:  BiocManager::install("GenVisR")
cnv-comparison/01-cnv-comparison-plotting.Rmd:  install.packages("cowplot")
molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd:  install.packages("DT")
molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd:  install.packages("htmlwidgets")
mutational-signatures/mutational_signatures.Rmd:  BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")
mutational-signatures/mutational_signatures.Rmd:  install.packages("deconstructSigs")
mutect2-vs-strelka2/01-set-up.Rmd:    install.packages("BiocManager")
mutect2-vs-strelka2/01-set-up.Rmd:  BiocManager::install("maftools")
mutect2-vs-strelka2/01-set-up.Rmd:  install.packages("hexbin")
mutect2-vs-strelka2/01-set-up.Rmd:  install.packages("R.utils")
mutect2-vs-strelka2/02-analyze-concordance.Rmd:  install.packages("hexbin")
mutect2-vs-strelka2/02-analyze-concordance.Rmd:  devtools::install_github("clauswilke/colorblindr")
mutect2-vs-strelka2/02-analyze-concordance.Rmd:  install.packages("R.utils")
mutect2-vs-strelka2/02-analyze-concordance.Rmd:  install.packages("VennDiagrams")
snv-callers/compare_snv_callers_plots-tcga.Rmd:  install.packages("ggupset", repos = "http://cran.us.r-project.org")
snv-callers/compare_snv_callers_plots-tcga.Rmd:  install.packages("UpSetR", repos = "http://cran.us.r-project.org")
snv-callers/compare_snv_callers_plots-tcga.Rmd:  install.packages("GGally", repos = "http://cran.us.r-project.org")
snv-callers/compare_snv_callers_plots.Rmd:  install.packages("ggupset", repos = "http://cran.us.r-project.org")
snv-callers/compare_snv_callers_plots.Rmd:  install.packages("UpSetR", repos = "http://cran.us.r-project.org")
snv-callers/compare_snv_callers_plots.Rmd:  install.packages("GGally", repos = "http://cran.us.r-project.org")
snv-callers/vaf_cutoff_experiment.Rmd:  install.packages("GGally")
snv-callers/vaf_cutoff_experiment.Rmd:  install.packages("ggupset")
```

### PR DESCRIPTION
**Pull request by cansavvy**
#### What was your approach?
I'm removing R package installations in the scripts in accordance with #690 

Here I've addressed what should be the last of the changes to wrap up and closes #690.

- Got rid of maftools install in `01-plot-oncoprint.R` and reran `run-oncoprint.sh`
- Get rid of package installations in `run-dimension-reducion.R` and reran `./dimension-reduction-plots.sh`
- Got rid of install in `util/survival_models.R` and reran `survival-analysis_template.Rmd`

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

- t-SNE numbers look different. Are we able to set the seed for that?

- Everything else look okay? Results that should be there are there? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes, but the results shouldn't really be different. 

#### Reproducibility Checklist
No changes are needed here. 

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
> t-SNE numbers look different. Are we able to set the seed for that?

PCA numbers also look slightly different. I am rerunning:
```
bash analyses/transcriptomic-dimension-reduction/dimension-reduction-plots.sh
```

In the Docker container on the branch for #675 right now and I've gotten through some of the steps that are showing changes here and nothing is showing up for me as altered in the scores TSV files.

Hardcoded seed in shell script `01-dimension-reduction.sh`: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/608f7c58e4987df819a9072c4fa22f6f8ac76616/analyses/transcriptomic-dimension-reduction/01-dimension-reduction.sh#L14

Here's that seed getting passed to `run-dimension-reduction.R`: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/608f7c58e4987df819a9072c4fa22f6f8ac76616/analyses/transcriptomic-dimension-reduction/01-dimension-reduction.sh#L28

Here's were we run t-SNE via a custom function in `scripts/run-dimension-reduction.R`: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/608f7c58e4987df819a9072c4fa22f6f8ac76616/analyses/transcriptomic-dimension-reduction/scripts/run-dimension-reduction.R#L221

Here's that seed getting set in the custom function: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/608f7c58e4987df819a9072c4fa22f6f8ac76616/analyses/transcriptomic-dimension-reduction/util/dimension-reduction-functions.R#L32

Wondering if something else is going on.

**Comment by jaclyn-taroni** [comment]
Ran the same thing on my `master` (which is even with AlexsLemonade `master`) and the only changes I get are to the PDF dates.

**Comment by cansavvy** [comment]
> Ran the same thing on my `master` (which is even with AlexsLemonade `master`) and the only changes I get are to the PDF dates.

@jaclyn-taroni I've checked out AlexsLemonade master and reran on the docker image and am still getting changed data. So I think we've narrowed down that it is due to changes in the docker image/container. Can you tell me more about what docker container/image you used? Specifically what version of Rtsne is it using? I'm gonna double check that I'm using the most updated Docker image as well.

**Comment by jaclyn-taroni** [comment]
The versions for Rtsne are unchanged as far as I know (the last release was end of 2018, I believe). I have a version of the Docker image that is post-refactor/change of the Bioconductor install process.

**Comment by cbethell** [comment]
Upon digging into this PR, I was unable to pinpoint a specific issue and/or solution but I did determine the following:  
- The seed function must be working as intended because my PCA and tSNE results stay the same (although they are different from the results in this PR and in master)
- The input files have not changed since the dimension reduction script was last ran and merged into master
- All other changes in this PR seem to be fine, the only issues appear to be with the PCA and tSNE result files
- The issue appears to be with the versions of the Rtsne and prcomp functions/packages included in Docker — with no evidence of them being updated (from CRAN)
- My results, however, were all only slightly off from the results in master

In other words, the mystery, unfortunately, still continues...

**Comment by jaclyn-taroni** [comment]
Can someone file an issue to track the t-SNE results not being reproducible (including how we might test if it's the underlying `prcomp` result changing)? And then we can revisit the rest of the changes here.

**Comment by cansavvy** [comment]
> I did notice that the `analyses/survival-analysis/results/cox_regression_tmb.tsv` output file also changed in this PR, with the conf.high variable within containing a change of 1.026 -> 7.275 (the most significant change I noticed).

Yes! Because of changes to tmb calculations (#564) I would expect the regression to have different results. I'll get those conflicts resolved.
